File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/external/erp/usecases/base/BaseErpUseCase.java
package com.optivem.shop.dsl.core.usecase.external.erp.usecases.base;

import com.optivem.shop.dsl.driver.port.external.erp.ErpDriver;
import com.optivem.shop.dsl.core.shared.BaseUseCase;
import com.optivem.shop.dsl.core.shared.UseCaseContext;

public abstract class BaseErpUseCase<TResponse, TVerification> extends BaseUseCase<ErpDriver, TResponse, TVerification> {
    protected BaseErpUseCase(ErpDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/External/Erp/UseCases/Base/BaseErpUseCase.cs
using Driver.Port.External.Erp;
using Driver.Port.External.Erp.Dtos.Error;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Erp.UseCases.Base;

public abstract class BaseErpUseCase<TResponse, TVerification>
    where TVerification : ResponseVerification<TResponse>
{
    protected readonly IErpDriver _driver;
    protected readonly UseCaseContext _context;

    protected BaseErpUseCase(IErpDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public abstract Task<ErpUseCaseResult<TResponse, TVerification>> Execute();
}



